From 751ef05c3f0f0585223728ae52a82078adba4989 Mon Sep 17 00:00:00 2001
From: yuan junhui <yuanjunhui@eswincomputing.com>
Date: Thu, 9 May 2024 11:17:11 +0800
Subject: [PATCH 013/109] WIN2030-15279:fix(bmc):add som,cd info,fix bug

Changelogs:
01,add boardinfo  sominfo,bcinfo
02,add format dip-switch redio
03,fixbug power on change ,disabled before getdata

Change-Id: I3dea439a8804722887c3931a59d74b934be21df3
---
 Core/Src/web-server.c | 285 ++++++++++++++++++++++++++++++++----------
 1 file changed, 222 insertions(+), 63 deletions(-)

diff --git a/Core/Src/web-server.c b/Core/Src/web-server.c
index 2d0cdd9..f74c84e 100644
--- a/Core/Src/web-server.c
+++ b/Core/Src/web-server.c
@@ -9630,14 +9630,10 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                             type: 'GET',\n \
 															contentType: 'application/x-www-form-urlencoded', \n \
                                                             success: function(response) {\n \
-                                                                $('#dip01').val(response.data.dip01);\n \
-                                                                $('#dip02').val(response.data.dip02);\n \
-                                                                $('#dip03').val(response.data.dip03);\n \
-																$('#dip04').val(response.data.dip04);\n \
-																$('#dip05').val(response.data.dip05);\n \
-																$('#dip06').val(response.data.dip06);\n \
-																$('#dip07').val(response.data.dip07);\n \
-																$('#dip08').val(response.data.dip08);\n \
+																$('input[name=\"dip01\"][value=\"' + response.data.dip01 + '\"]').prop('checked', true);\n \
+																$('input[name=\"dip02\"][value=\"' + response.data.dip02 + '\"]').prop('checked', true);\n \
+																$('input[name=\"dip03\"][value=\"' + response.data.dip03 + '\"]').prop('checked', true);\n \
+																$('input[name=\"dip04\"][value=\"' + response.data.dip04 + '\"]').prop('checked', true);\n \
                                                                 console.log('dip_switch refreshed successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
@@ -9647,14 +9643,10 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                         });\n\
                                                     });\n\
 													$('#dip-switch-update').click(function() { \n \
-                                                        var dip01 = $('#dip01').val();\n \
-                                                        var dip02 = $('#dip02').val();\n \
-                                                        var dip03 = $('#dip03').val();\n \
-														var dip04 = $('#dip04').val();\n \
-														var dip05 = $('#dip05').val();\n \
-														var dip06 = $('#dip06').val();\n \
-														var dip07 = $('#dip07').val();\n \
-														var dip08 = $('#dip08').val();\n \
+                                                        var dip01 = $('input[name=\"dip01\"]:checked').val(); \n \
+														var dip02 = $('input[name=\"dip02\"]:checked').val(); \n \
+														var dip03 = $('input[name=\"dip03\"]:checked').val(); \n \
+														var dip04 = $('input[name=\"dip04\"]:checked').val(); \n \
                                                         $.ajax({ \n\
                                                             url: '/dip_switch',\n \
                                                             type: 'POST',\n \
@@ -9663,11 +9655,7 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                                 dip01: dip01, \n \
                                                                 dip02: dip02, \n \
                                                                 dip03: dip03, \n \
-																dip04: dip04, \n \
-																dip05: dip05, \n \
-																dip06: dip06, \n \
-																dip07: dip07, \n \
-																dip08: dip08  \n \
+																dip04: dip04 \n \
                                                             },\
                                                             success: function(response) {\n \
                                                                 if(response.status===0){\n \
@@ -9686,6 +9674,9 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                         });\n \
                                                     });\n \
 													$('#power-on-refresh').click(function() {\n \
+														$('#power-on-status').val('loading,,,');\n \
+														$('#power-on-change').prop(\"disabled\", true); \n \
+														$('#power-on-change').text('loading,,,');\n \
                                                         $.ajax({\n \
 															async: false,\n \
                                                             url: '/power_status',\n \
@@ -9693,6 +9684,7 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
 															contentType: 'application/x-www-form-urlencoded', \n \
 															success: function(response) {\n \
 																// 假设返回的数据是{ buttonText: '新按钮文本' }\n \
+																$('#power-on-change').prop(\"disabled\", false); \n \
 																if(response.data.power_status===\"0\"){\n \
 																	$('#power-on-status').val('powerOFF');\n \
 																	$('#power-on-change').text('powerON');\n \
@@ -9709,12 +9701,16 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                         });\n\
                                                     });\n\
 													$('#power-retention-refresh').click(function() {\n \
+														$('#power-retention-status').val('loading,,,');\n \
+														$('#power-retention-change').prop(\"disabled\", true); \n \
+														$('#power-retention-change').text('loading,,,');\n \
                                                         $.ajax({\n \
 															async: false,\n \
                                                             url: '/power_retention_status',\n \
 															type: 'GET',\n \
 															contentType: 'application/x-www-form-urlencoded', \n \
 															success: function(response) {\n \
+																$('#power-retention-change').prop(\"disabled\", false); \n \
 																if(response.data.power_retention_status===\"0\"){\n \
 																	$('#power-retention-status').val('disabled');\n \
 																	$('#power-retention-change').text('enable');\n \
@@ -9729,7 +9725,45 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
 																console.error('An error occurred:', error);\n \
 															}\n \
                                                         });\n\
-                                                    });\n\
+                                                    });\n \
+													$('#board-info-som-refresh').click(function() {\n \
+                                                        $.ajax({\n \
+															async: false,\n \
+                                                            url: '/board_info_som',\n \
+                                                            type: 'GET',\n \
+                                                            success: function(response) {\n \
+                                                                $('#som_magicNumber').val(response.data.magicNumber);\n \
+                                                                $('#som_formatVersionNumber').val(response.data.formatVersionNumber);\n \
+                                                                $('#som_productIdentifier').val(response.data.productIdentifier);\n \
+																$('#som_pcbRevision').val(response.data.pcbRevision);\n \
+																$('#som_boardSerialNumber').val(response.data.boardSerialNumber);\n \
+                                                                console.log('board-info-som-refresh refreshed successfully.');\n \
+                                                            },\n \
+                                                            error: function(xhr, status, error) {\n \
+                                                                // 请求失败时的回调函数\n \
+                                                                console.error('Error refreshing board-info-som-refresh:', error);\n \
+                                                            }\n \
+                                                        });\n \
+                                                    });\n \
+													$('#board-info-cb-refresh').click(function() {\n \
+                                                        $.ajax({\n \
+															async: false,\n \
+                                                            url: '/board_info_cb',\n \
+                                                            type: 'GET',\n \
+                                                            success: function(response) {\n \
+                                                                $('#cb_magicNumber').val(response.data.magicNumber);\n \
+                                                                $('#cb_formatVersionNumber').val(response.data.formatVersionNumber);\n \
+                                                                $('#cb_productIdentifier').val(response.data.productIdentifier);\n \
+																$('#cb_pcbRevision').val(response.data.pcbRevision);\n \
+																$('#cb_boardSerialNumber').val(response.data.boardSerialNumber);\n \
+                                                                console.log('board-info-cb-refresh refreshed successfully.');\n \
+                                                            },\n \
+                                                            error: function(xhr, status, error) {\n \
+                                                                // 请求失败时的回调函数\n \
+                                                                console.error('Error refreshing board-info-cb-refresh:', error);\n \
+                                                            }\n \
+                                                        });\n \
+                                                    });\n \
 													// --------------------after load page ------------- \n \
 													$('#power-on-refresh').click(); \n \
 													$('#power-retention-refresh').click(); \n \
@@ -9737,6 +9771,8 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
 													$('#pvt-info-refresh').click(); \n \
 													$('#dip-switch-refresh').click(); \n \
    													$('#net-work-refresh').click(); \n \
+													$('#board-info-som-refresh').click(); \n \
+													$('#board-info-cb-refresh').click(); \n \
                                                 });\n\
                                             </script>\n \
                                             <style> \
@@ -9769,20 +9805,57 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
 												<button type=\"submit\">Exit</button> \
 											</form> \
                                         </div> \
+										<div> \
+											<h3>Board Information</h3> \
+											<h4>SOM Info</h4> \
+											<div class=\"network-row\"> \
+											<label>magicNumber:</label> <input type=\"text\" id=\"som_magicNumber\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											</div> \
+											<div class=\"network-row\"> \
+											<label>formatVersionNumber:</label> <input type=\"text\" id=\"som_formatVersionNumber\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											</div> \
+											<div class=\"network-row\"> \
+											<label>productIdentifier:</label> <input type=\"text\" id=\"som_productIdentifier\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											</div> \
+											<div class=\"network-row\"> \
+											<label>pcbRevision:</label> <input type=\"text\" id=\"som_pcbRevision\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											</div> \
+											<div class=\"network-row\"> \
+											<label>boardSerialNumber:</label> <input type=\"text\" id=\"som_boardSerialNumber\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											</div> \
+											<button id=\"board-info-som-refresh\" >refresh</button> \
+											<h4>CarrierBoard Info</h4> \
+											<div class=\"network-row\"> \
+											<label>magicNumber:</label> <input type=\"text\" id=\"cb_magicNumber\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											</div> \
+											<div class=\"network-row\"> \
+											<label>formatVersionNumber:</label> <input type=\"text\" id=\"cb_formatVersionNumber\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											</div> \
+											<div class=\"network-row\"> \
+											<label>productIdentifier:</label> <input type=\"text\" id=\"cb_productIdentifier\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											</div> \
+											<div class=\"network-row\"> \
+											<label>pcbRevision:</label> <input type=\"text\" id=\"cb_pcbRevision\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											</div> \
+											<div class=\"network-row\"> \
+											<label>boardSerialNumber:</label> <input type=\"text\" id=\"cb_boardSerialNumber\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											</div> \
+											<button id=\"board-info-cb-refresh\" >refresh</button> \
+										</div> \
                                        <h3>Machine Status</h3> \
 										<div class=\"power-status\"> \
 											<div class=\"network-row\"> \
-											<label>Power Status:</label> <input type=\"text\" id=\"power-on-status\" value=\"0\" style=\"width: 100px;\" disabled>  <br> \
+											<label>Power Status:</label> <input type=\"text\" id=\"power-on-status\" value=\"loading,,,\" style=\"width: 100px;\" disabled>  <br> \
 											</div> \
 											<div class=\"network-row\"> \
-											<label>Power Status Change:</label> <button id=\"power-on-change\" >unknow</button>   <br> \
+											<label>Power Status Change:</label> <button id=\"power-on-change\" disabled>loading,,,</button>   <br> \
 											</div> \
 											<button id=\"power-on-refresh\" style=\"display:none;\">refresh</button> \
 											<div class=\"network-row\"> \
-											<label>Power Retention:</label> <input type=\"text\" id=\"power-retention-status\" value=\"0\" style=\"width: 100px;\" disabled> <br> \
+											<label>Power Retention:</label> <input type=\"text\" id=\"power-retention-status\" value=\"loading,,,\" style=\"width: 100px;\" disabled> <br> \
 											</div> \
 											<div class=\"network-row\"> \
-											<label>Power Retention Change:</label> <button id=\"power-retention-change\" >unknow</button> <br> \
+											<label>Power Retention Change:</label> <button id=\"power-retention-change\" disabled>loading,,,</button> <br> \
 											</div> \
 											<button id=\"power-retention-refresh\" style=\"display:none;\">refresh</button> \
 										</div> \
@@ -9819,14 +9892,26 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                         </div> \
 										 <div class=\"dip-switch\" > \
                                             <h3>DIP Switch</h3> \
-                                            dip01:<input  type=\"text\" id=\"dip01\" style=\"width: 60px;\" disabled> <br> \
-											dip02:<input  type=\"text\" id=\"dip02\" style=\"width: 60px;\" disabled> <br> \
-											dip03:<input  type=\"text\" id=\"dip03\" style=\"width: 60px;\" > <br> \
-											dip04:<input  type=\"text\" id=\"dip04\" style=\"width: 60px;\" > <br> \
-											dip05:<input  type=\"text\" id=\"dip05\" style=\"width: 60px;\" > <br> \
-											dip06:<input  type=\"text\" id=\"dip06\" style=\"width: 60px;\" > <br> \
-											dip07:<input  type=\"text\" id=\"dip07\" style=\"width: 60px;\" > <br> \
-											dip08:<input  type=\"text\" id=\"dip08\" style=\"width: 60px;\" > <br> \
+                                           <div  class=\"network-row\"> \
+												<label>dip01:</label> \
+												<input type=\"radio\" name=\"dip01\" value=\"0\" id=\"dip01-false\"><label for=\"dip01-false\">close</label> \
+												<input type=\"radio\" name=\"dip01\" value=\"1\" id=\"dip01-true\"><label for=\"dip01-true\">open</label> \
+											</div> \
+											<div  class=\"network-row\"> \
+												<label>dip02:</label> \
+												<input type=\"radio\" name=\"dip02\" value=\"0\" id=\"dip02-false\"><label for=\"dip02-false\">close</label> \
+												<input type=\"radio\" name=\"dip02\" value=\"1\" id=\"dip02-true\"><label for=\"dip02-true\">open</label> \
+											</div> \
+											<div  class=\"network-row\"> \
+												<label>dip03:</label> \
+												<input type=\"radio\" name=\"dip03\" value=\"0\" id=\"dip03-false\"><label for=\"dip03-false\">close</label> \
+												<input type=\"radio\" name=\"dip03\" value=\"1\" id=\"dip03-true\"><label for=\"dip03-true\">open</label> \
+											</div> \
+											<div  class=\"network-row\"> \
+												<label>dip04:</label> \
+												<input type=\"radio\" name=\"dip04\" value=\"0\" id=\"dip04-false\"><label for=\"dip04-false\">close</label> \
+												<input type=\"radio\" name=\"dip04\" value=\"1\" id=\"dip04-true\"><label for=\"dip04-true\">open</label> \
+											</div> \
                                             <button id=\"dip-switch-refresh\">refresh</button> <button id=\"dip-switch-update\">update</button> \
                                         </div> \
                                         <div class=\"net-work\" > \
@@ -9843,7 +9928,7 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
 											 <div class=\"network-row\"> \
                                             <label>Subnet Mask:</label> <input type=\"text\" id=\"subnetwork\" value=\"0\" style=\"width: 180px;\"> <br> \
 											 </div> \
-                                            <button id=\"net-work-update\">update</button> <button id=\"net-work-refresh\">refresh</button> \
+                                            <button id=\"net-work-refresh\">refresh</button> <button id=\"net-work-update\">update</button> \
                                         </div> \
                                     </body> \
                                     </html> ";       
@@ -10355,10 +10440,6 @@ typedef struct {
     int dip02;
     int dip03;
 	int dip04;
-	int dip05;
-	int dip06;
-	int dip07;
-	int dip08;
 } DIPSwitchInfo;
 
 DIPSwitchInfo get_dip_switch(){
@@ -10366,12 +10447,8 @@ DIPSwitchInfo get_dip_switch(){
 	DIPSwitchInfo dipSwitchInfo;
 	dipSwitchInfo.dip01 = 0;
 	dipSwitchInfo.dip02 = 0;
-	dipSwitchInfo.dip03 = 0;
-	dipSwitchInfo.dip04 = 0;
-	dipSwitchInfo.dip05 = 1;
-	dipSwitchInfo.dip06 = 1;
-	dipSwitchInfo.dip07 = 1;
-	dipSwitchInfo.dip08 = 1;
+	dipSwitchInfo.dip03 = 1;
+	dipSwitchInfo.dip04 = 1;
     return dipSwitchInfo;
 }
 int set_dip_switch(DIPSwitchInfo dipSwitchInfo){
@@ -10379,6 +10456,50 @@ int set_dip_switch(DIPSwitchInfo dipSwitchInfo){
     return 0;//0:success ,other:error
 }
 
+
+
+typedef struct {
+	int magicNumber;
+	int formatVersionNumber;
+	int productIdentifier;
+	int pcbRevision;
+	char boardSerialNumber[18];
+} SOMSimpleInfo;
+
+
+SOMSimpleInfo get_som_info(){
+	printf("TODO call get_som_info\n");
+	SOMSimpleInfo example= {
+        0,
+        1,
+        2,
+        3,
+		"v1.10.1"
+    };
+    return example;
+}
+
+typedef struct {
+	int magicNumber;
+	int formatVersionNumber;
+	int productIdentifier;
+	int pcbRevision;
+	char boardSerialNumber[18];
+} CBSimpleInfo;//carrie board
+
+CBSimpleInfo get_cb_info(){
+	printf("TODO call get_cb_info\n");
+	CBSimpleInfo example= {
+        0,
+        1,
+        2,
+        3,
+		"v1.10.1"
+    };
+    return example;
+}
+
+
  /** Serve one HTTP connection accepted in the http thread */
  static void
  http_server_netconn_serve(struct netconn *conn)
@@ -10639,10 +10760,10 @@ int set_dip_switch(DIPSwitchInfo dipSwitchInfo){
 				printf("GET location: dip_switch \n");
 				DIPSwitchInfo dipSwitchInfo= get_dip_switch();
 
-				char json_response[BUF_SIZE_256]={0};
+				char json_response[BUF_SIZE_128]={0};
 				 // 创建JSON格式的字符串
-                char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"dip01\":\"%d\",\"dip02\":\"%d\",\"dip03\":\"%d\",\"dip04\":\"%d\",\"dip05\":\"%d\",\"dip06\":\"%d\",\"dip07\":\"%d\",\"dip08\":\"%d\"}}";
-                sprintf(json_response, json_response_patt,dipSwitchInfo.dip01,dipSwitchInfo.dip02,dipSwitchInfo.dip03,dipSwitchInfo.dip04,dipSwitchInfo.dip05,dipSwitchInfo.dip06,dipSwitchInfo.dip07,dipSwitchInfo.dip08);
+                char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"dip01\":\"%d\",\"dip02\":\"%d\",\"dip03\":\"%d\",\"dip04\":\"%d\"}}";
+                sprintf(json_response, json_response_patt,dipSwitchInfo.dip01,dipSwitchInfo.dip02,dipSwitchInfo.dip03,dipSwitchInfo.dip04);
 
                 // 发送HTTP头部
                 const char *header = "HTTP/1.1 200 OK\r\n"
@@ -10720,12 +10841,62 @@ int set_dip_switch(DIPSwitchInfo dipSwitchInfo){
 
 			
                 char response_header[256];
-                sprintf(response_header, http_jquery_200_patt, strlen(data_jquery_min_js)-91);
+                sprintf(response_header, http_jquery_200_patt, strlen(data_jquery_min_js)-90);
 
                 printf("strlen(http_jquery_200) %d \n",strlen(response_header));
-                printf("strlen(data_jquery_min_js) %d \n", strlen(data_jquery_min_js)-91);
+                printf("strlen(data_jquery_min_js) %d \n", strlen(data_jquery_min_js)-90);
+                netconn_write(conn, response_header, strlen(response_header), NETCONN_COPY);
+                netconn_write(conn, data_jquery_min_js, strlen(data_jquery_min_js)-90, NETCONN_COPY);
+
+            }else if(strncmp(path, "/board_info_som",15)==0 ){ 
+                printf("get ,location: /board_info_som \n");
+				SOMSimpleInfo simpleInfo=get_som_info();
+// magicNumber
+// formatVersionNumber
+// productIdentifier
+// pcbRevision
+// boardSerialNumber
+				char json_response[BUF_SIZE_256]={0};
+				 // 创建JSON格式的字符串
+                char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"magicNumber\":\"%d\",\"formatVersionNumber\":\"%d\",\"productIdentifier\":\"%d\",\"pcbRevision\":\"%d\",\"boardSerialNumber\":\"%s\"}}";
+                sprintf(json_response, json_response_patt,simpleInfo.magicNumber,simpleInfo.formatVersionNumber,simpleInfo.productIdentifier,simpleInfo.pcbRevision,simpleInfo.boardSerialNumber);
+
+                // 发送HTTP头部
+                const char *header = "HTTP/1.1 200 OK\r\n"
+                                    "Content-Type: application/json\r\n"
+                                    "Connection: close\r\n"
+                                    "Content-Length: %d\r\n\r\n";
+
+                char response_header[BUF_SIZE_512];
+                sprintf(response_header, header, strlen(json_response));
+
+                printf("strlen(response_header):%d,strlen(json_response):%d \n",strlen(response_header),strlen(json_response));
+
                 netconn_write(conn, response_header, strlen(response_header), NETCONN_COPY);
-                netconn_write(conn, data_jquery_min_js, strlen(data_jquery_min_js)-91, NETCONN_COPY);
+                netconn_write(conn, json_response, strlen(json_response), NETCONN_COPY);
+
+            }else if(strncmp(path, "/board_info_cb",14)==0 ){ 
+                printf("get ,location: /board_info_cb \n");
+				CBSimpleInfo simpleInfo=get_cb_info();
+                
+               	char json_response[BUF_SIZE_256]={0};
+				 // 创建JSON格式的字符串
+                char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"magicNumber\":\"%d\",\"formatVersionNumber\":\"%d\",\"productIdentifier\":\"%d\",\"pcbRevision\":\"%d\",\"boardSerialNumber\":\"%s\"}}";
+                sprintf(json_response, json_response_patt,simpleInfo.magicNumber,simpleInfo.formatVersionNumber,simpleInfo.productIdentifier,simpleInfo.pcbRevision,simpleInfo.boardSerialNumber);
+
+                // 发送HTTP头部
+                const char *header = "HTTP/1.1 200 OK\r\n"
+                                    "Content-Type: application/json\r\n"
+                                    "Connection: close\r\n"
+                                    "Content-Length: %d\r\n\r\n";
+
+                char response_header[BUF_SIZE_512];
+                sprintf(response_header, header, strlen(json_response));
+
+                printf("strlen(response_header):%d,strlen(json_response):%d \n",strlen(response_header),strlen(json_response));
+
+                netconn_write(conn, response_header, strlen(response_header), NETCONN_COPY);
+                netconn_write(conn, json_response, strlen(json_response), NETCONN_COPY);
 
             }else{
                 printf("ERROR unsupport get path \n");
@@ -11099,18 +11270,6 @@ int set_dip_switch(DIPSwitchInfo dipSwitchInfo){
                         }else if(strcmp(current->key,"dip04")==0){
                             intValue = strtol(current->value, &endPtr, 10);
                             dipSwitchInfo.dip04= intValue>0?1:0;
-                        }else if(strcmp(current->key,"dip05")==0){
-                            intValue = strtol(current->value, &endPtr, 10);
-                            dipSwitchInfo.dip05= intValue>0?1:0;
-                        }else if(strcmp(current->key,"dip06")==0){
-                            intValue = strtol(current->value, &endPtr, 10);
-                            dipSwitchInfo.dip06= intValue>0?1:0;
-                        }else if(strcmp(current->key,"dip07")==0){
-                            intValue = strtol(current->value, &endPtr, 10);
-                            dipSwitchInfo.dip07= intValue>0?1:0;
-                        }else if(strcmp(current->key,"dip08")==0){
-                            intValue = strtol(current->value, &endPtr, 10);
-                            dipSwitchInfo.dip08= intValue>0?1:0;
                         }
                         current = current->next;
                     }
-- 
2.25.1

